rusk-wallet/src/lib/prompt.rs

```diff
@@ -126,121 +136,197 @@ pub(crate) fn welcome() -> u8 {
+pub(crate) fn select_wallet(dir: &str, wallets: Vec<String>) -> Option<PathBuf> {
     // select the wallet
     let q = Question::select("wallet")
         .message("Please select the wallet you wish to use:")
         .choices(&wallets)
+        .default_separator()
+        .choice("Other...")
         .build();
     let a = requestty::prompt_one(q).unwrap();
     let wi = a.as_list_item().unwrap().index;
 
-    // gen full path for selected wallet
-    let mut path = PathBuf::new();
-    path.push(dir);
-    path.push(wallets[wi].clone());
-    path
+    if wi > wallets.len() {
+        None
+    } else {
+        // gen full path for selected wallet
+        let mut path = PathBuf::new();
+        path.push(dir);
+        path.push(wallets[wi].clone());
+        Some(path)
+    }
+
+}
+
+/// Request a name for the wallet
+pub(crate) fn request_wallet_name() -> String {
+
+    let q = Question::input("name")
+        .message("Please enter a wallet name:")
+        .validate_on_key(|name, _| !wallet_exists(name))
+        .validate(|name, _| {
+            if !wallet_exists(name) {
+                Ok(())
+            } else {
+                Err("A wallet with this name already exists".to_string())
+            }
+        })
+        .build();
+
+    let a = requestty::prompt_one(q).unwrap();
+    a.as_string().unwrap().to_string()
+}
+
+/// Checks if a wallet with this name already exists
+fn wallet_exists(name: &str) -> bool {
+    let mut pb = PathBuf::new();
+    pb.push(WalletCfg::default_data_dir());
+    pb.push(name);
+    pb.set_extension("dat");
+    pb.is_file()
 }
 
 /// Let the user choose a command to execute
-pub(crate) fn command() -> Option<CliCommand> {
-    let msg = "What would you like to do?";
+pub(crate) fn command(offline: bool) -> Option<CliCommand> {
+
+    // notify the user if we're note connected
+    let offline_notice = match offline {
+        false => "",
+        true => " [offline]",
+    };
+
+    // choices differ if user is online or not
+    let choices = match offline {
+        false => {
+            vec![
+                "Retrieve my public spend key",
+                "Check my current balance",
+                "Send Dusk",
+                "Stake Dusk",
+                "Unstake Dusk",
+            ]
+        }
+        true => {
+            vec![
+                "Retrieve my public spend key",
+            ]
+        }
+    };
+
+
+    let msg = format!("What would you like to do?{}", offline_notice);
     let q = Question::select("action")
         .message(msg)
-        .choices(vec![
-            "Check my current balance",
-            "Retrieve my public spend key",
-            "Send Dusk",
-            "Stake Dusk",
-            "Extend stake for a particular key",
-            "Withdraw a key's stake",
-            "Export provisioner BLS key pair",
-        ])
+        .choices(choices)
+        .default_separator()
+        .choice("Export provisioner BLS key pair")
```

Comment:
Any particular reason to handle this choice in a different way?

Reply:
It falls under a different "command category" imho, I felt it was better separated from the others.